### ISSUE TITLE
Highlight classes names and methods names?

### ISSUE THREAD

**Issue by harg** [leafgarland/typescript-vim#119]
Is it possible to highlight classes names and methods names?

![_no_name____-_vim2](https://user-images.githubusercontent.com/34398/29571679-2189c1dc-875a-11e7-8a1a-57349e32d028.png)

**Comment by leafgarland** [comment]
Yes it is possible but only by adding new syntax matches to this syntax. Essentially that means defining regex patterns that will match those class and method names.

**Comment by Zabanaa** [comment]
Do you know any resources on how to create syntax files ? I'd love to take care of this

**Comment by leafgarland** [comment]
Hi @Zabanaa, a good place to start with any vimscript learnings is Steve Losh's [Learn Vimscript the Hardway](http://learnvimscriptthehardway.stevelosh.com/). There are two chapters in there on syntax highlighting. Also the built-in help, the user-guide `:help usr_44` and reference `:help syn-define`.

I would be very happy to get more contributions on this, thanks.

**Comment by harg** [comment]
@Zabanaa maybe you can check https://github.com/pangloss/vim-javascript for inspiration

**Comment by tommyc38** [comment]
vim's built in typescript syntax highlighter does this.  The problem I had with the native solution was that it breaks on template strings (e.g. backticks)...which is why I tried out this plugin.    I also notice that `=|!|false|true` are different colors than what the theming states they should be. This essentially breaks any native themes.  There has to be something that isn't mapped correctly.
